pkg/executor/infoschema_reader.go

```diff
@@ -3498,6 +3499,37 @@ func (e *TiFlashSystemTableRetriever) dataForTiFlashSystemTables(ctx context.Con
 			}
 		}
 		outputRow[len(e.outputCols)-1].SetString(instanceID, mysql.DefaultCollationName)
+
+		// for "tiflash_indexes", set the column_name and index_name according to the TableInfo
+		if e.table.Name.L == "tiflash_indexes" {
+			var logicalTableID = outputRow[outputColIndexMap["table_id"]].GetInt64()
+			if !outputRow[outputColIndexMap["belonging_table_id"]].IsNull() {
+				// Old TiFlash versions may not have this column. In this case we will try to get by the "table_id"
+				belongingTableID := outputRow[outputColIndexMap["belonging_table_id"]].GetInt64()
+				if belongingTableID != -1 && belongingTableID != 0 {
+					logicalTableID = belongingTableID
+				}
+			}
+			if table, ok := is.TableByID(ctx, logicalTableID); ok {
+				tableInfo := table.Meta()
+				getInt64DatumVal := func(datum_name string, default_val int64) int64 {
+					datum := outputRow[outputColIndexMap[datum_name]]
+					if !datum.IsNull() {
+						return datum.GetInt64()
+					}
+					return default_val
+				}
+				// set column_name
+				columnID := getInt64DatumVal("column_id", 0)
```

Comment:
Could we remove `column_id` in this repo?

Reply:
Removing `column_id` will make it hard to fetch the "column_id" from tiflash's `system.dt_local_index` table, and then it make more difficult to identify which column_name the index is built on. According to the previous discussion with @breezewish , we'd better provide the column_name for usability. So I'd like to keep the `column_id` field in the `information_schema.tiflash_indexes` table.